grommet/grommet
# Fix DataTable groupBy.onSelect condition

#### What does this PR do?
This fixes a condition to check if `groupBy` prop exists before trying to access `groupBy.onSelect` and restores `onSelect` functionality for DataTables that don't have `groupBy.onSelect`.

Our jest tests did not have a test for this, so I have added one. I confirmed that the test failed with the published code and is resolved with the fix in this PR.

#### Where should the reviewer start?
src/js/components/DataTable/Header.js

#### What testing has been done on this PR?
Tested `onSelect` storybook and added jest test.

#### How should this be manually tested?
`onSelect` storybook, click the checkbox in the header to select all, all should be selected.

#### Any background context you want to provide?

#### What are the relevant issues?

#### Screenshots (if appropriate)

BEFORE:

<img width="949" alt="Screen Shot 2021-11-20 at 7 45 18 AM" src="https://user-images.githubusercontent.com/12522275/142691025-48d43328-b938-490b-acfd-88d628090a03.png">

<img width="386" alt="Screen Shot 2021-11-20 at 7 45 09 AM" src="https://user-images.githubusercontent.com/12522275/142691029-472a9551-0723-4b7e-9385-eb0efc4a7db5.png">

AFTER:
<img width="343" alt="Screen Shot 2021-11-20 at 8 02 23 AM" src="https://user-images.githubusercontent.com/12522275/142691256-9b643da7-f6b3-4041-b0f8-faf25c7d64fa.png">

#### Do the grommet docs need to be updated?
no.

#### Should this PR be mentioned in the release notes?
Yes. Fixed DataTable `onSelect` bug.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
